candies/cli/arg.py

```diff
@@ -0,0 +1,12 @@
+class Arg:
+    """An argument of a command line.
```

Comment:
I feel like it'd be better to add a little explanation here about the usage of the `Arg` class.

For example:
```py
class Arg:
    """An argument of a command line.
       
    Intended to be used as a type annotation in CLI functions
    to help the parser distinguish simple arguments, flags, prompts, etc.

    Note:
        Consider looking at the documentation
        of the `candies.cli.CLI` class for more details about
        which kinds of CLIs the `Arg` produces.

    Examples:
        @cli
        def main(a: Arg['argument a'],
                 b: Arg['argument b']):
            ...

    ...
    """
```

Reply:
Totally agree! Updated.